File: Creational/Builder/JavaScript/js/Meal.js
// js/Meal.js
export class Meal {
    constructor() {
        this.mainItem = '';
        this.sideItem = '';
        this.drink = '';
        this.toy = '';
    }

    setMainItem(mainItem) {
        this.mainItem = mainItem;
    }

    setSideItem(sideItem) {
        this.sideItem = sideItem;
    }

    setDrink(drink) {
        this.drink = drink;
    }

    setToy(toy) {
        this.toy = toy;
    }

    describe() {
        return `Meal [Main Item: ${this.mainItem}, Side Item: ${this.sideItem}, Drink: ${this.drink}, Toy: ${this.toy}]`;
    }
}

File: Creational/Builder/GoLang/meal.go
package main

import "fmt"

type Meal struct {
	MainItem string
	SideItem string
	Drink    string
	Toy      string
}

func (m *Meal) Describe() string {
	return fmt.Sprintf("Meal [Main Item: %s, Side Item: %s, Drink: %s, Toy: %s]", m.MainItem, m.SideItem, m.Drink, m.Toy)
}
